python/tests/test_file_echo_api.py

```diff
@@ -537,3 +538,102 @@ def test_occupancy_and_state_destruction(self):
         expected = {'error':{'data':{'stdout':None,'stderr':None},'code':22,'message':'Server at max capacity'},'jsonrpc':'2.0','id':uid3}
         self.assertEqual(actual3, expected)
 
+
+
+
+class InterruptTests(unittest.TestCase):
+    # Connection to server
+    c = None
+    # process running the server
+    p = None
+
+    @classmethod
+    def setUpClass(self):
+        p = subprocess.Popen(
+            ["cabal", "run", "exe:file-echo-api", "--verbose=0", "--", "--max-occupancy", "2", "http", "/", "--port", "8083", "--file", str(hello_file)],
+            stdout=subprocess.PIPE,
+            stdin=subprocess.DEVNULL,
+            stderr=subprocess.PIPE,
+            start_new_session=True)
+        time.sleep(3)
+        assert(p is not None)
+        poll_result = p.poll()
+        if poll_result is not None:
+            print(poll_result)
+            print(p.stdout.read())
+            print(p.stderr.read())
+        assert(poll_result is None)
+
+        self.p = p
+
+    @classmethod
+    def tearDownClass(self):
+        os.killpg(os.getpgid(self.p.pid), signal.SIGKILL)
+        super().tearDownClass()
+
+    # to be implemented by classes extending this one
+    def test_interrupts(self):
+        c1 = argo.ServerConnection(argo.HttpProcess('http://localhost:8083/'))
+        c2 = argo.ServerConnection(argo.HttpProcess('http://localhost:8083/'))
```

Comment:
Can the `argo.HttpProcess('http://localhost:8083/')` bit be shared between the two lines above?

Reply:
I don't think so - the HTTP requests are blocking, so to perform interrupts we need two independent connections.

Reply:
To be clear, I'm not suggesting sharing the same connection, just the URL that each one connects to. I had something like this in mind:

```py
proc = argo.HttpProcess('http://localhost:8083/')
c1 = argo.ServerConnection(proc)
c2 = argo.ServerConnection(proc)
```

(Disclaimer: I haven't tested this.)

Reply:
Unfortunately it's more than the URL contained there, i.e., `HttpProcess` has mutable state that can't be shared between `c1` and `c2` without breaking things. E.g., the `waiting_replies` member field.